src/component/jhi-boolean-badge.component.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ Copyright 2013-2017 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://jhipster.github.io/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+import { Component, Input, OnInit } from '@angular/core';
+import { JhiConfigService } from '../config.service';
+@Component({
+    selector: 'jhi-boolean-badge',
+    template: `<span *ngIf="i18nEnabled"
+                   class="badge"
+                   [ngClass]="clazz"
+                   [jhiTranslate]="translationKey">
+               </span>
+               <span *ngIf="!i18nEnabled"
+                   class="badge"
+                   [ngClass]="clazz">{{translationKey}}
```

Comment:
I just have one more problem its weird to use translationKeyTrue/translationKeyFalse.
I can suggest a better option, leave translation out of this component and when you use the component call it using the translate pipe, something like
`<jhi-boolean-badge [value]="value" trueText"my.i18n.trueKey | translate" falseText"my.i18n.falseKey | translate" />`
This way you can decouple i18n from this component which is much better IMO